tiledb/sm/filesystem/vfs.h

```diff
@@ -412,6 +414,230 @@ class VFS {
     std::vector<std::tuple<uint64_t, void*, uint64_t>> regions;
   };
 
+  /**
+   * Manages the read-ahead buffer cache for a set of LRU URIs.
+   */
+  class ReadAheadCache {
```

Comment:
Any chance we could reuse the existing `LRUCache` class?

Reply:
Oh, thanks. I didn't see that, it will probably work. Let me check.

Reply:
Yeah, sorry for the trouble. You may still need to wrap it for your purposes, but it may be good to reuse whatever functionality we can.

Reply:
I agree -- I may just template the LRUCache as necessary to support both this and the existing `StorageManager::tile_cache_` (which appears to be the only user of LRUCache.

Reply:
@stavrospapadopoulos I've given this a closer look: While I could modify the existing `LRUCache` to support generic datatypes to fit my use-case, there is a non-trivial issue with the existing `LRUCache`:

The `LRUCache::read` makes a copy. This makes sense because the `LRUCache` owns the objects. For my use-case, I do not need a copy of the entire object (an entire read-ahead buffer). I need a copy of a subrange. For example, I may store a buffer in the LRU with a range [0, 1000) and then make reads of [0, 100), [100, 200), ... [900, 1000). Using the existing `LRUCache::read` interface would force me to copy the entire [0, 1000) for each hit and then make a sub-copy to return to the user.

This seems like a problem tightly coupled with a read-ahead cache that it makes more sense to keep this tiny LRU here rather than trying to reuse the existing `LRUCache`.

Reply:
Done -- the `ReadAheadCache` now inherits from `LRUCache`.